### DISCUSSION TITLE
Its possible to estimate how long will take?

### DISCUSSION THREAD

**Discussion by LuccaPrado** [firegloves/MemPOI#11] [Q&A]
Theres some way to estimate how long mempoi will take to generate the sheet? Or something like percentage of the progress?

**Comment by firegloves** [comment]
Hi there!

nice question. Currently it's not possible, but it could be a good improvement.
I have to check how it will impact performances and probably we could only calculate the percentage. Something similar to

`resultSet total records / processed records`

Do you need this behavior because you are experiencing poor performances or because you need to process large datasets and you want your users to have feedback?

**Reply by LuccaPrado** [reply]
In fact it was just curiosity 😅

I would like to give an feedback to the user, but, today I don't have a problem on saying that when his Excel is ready I'll email it.

**Reply by firegloves** [reply]
Ok, so we could keep it in the backlog. Can I ask you how much time your medium export takes generally?

**Reply by LuccaPrado** [reply]
The query was very complex but it still export in about 1 minute.

**Reply by firegloves** [reply]
Could be an option the query tuning? I'm asking this because the only way to add completion percentage that I can imagine is to add a call in the core MemPOI cycle. So I mean that for each ResultSet record we could recalculate the completion percentage and set it in a separate (maybe static due to the async approach) class and on very large ResultSet it could slow down the entire process. However I can easily test it and observe how much time it would cost.

**Reply by LuccaPrado** [reply]
@firegloves, As I said, the question isn't about my case, it was just for curiosity, as it could be an nice feature to help the lib gain more users.

In my case, the speed of the export is really very very good in my opinion.

**Reply by firegloves** [reply]
Ok understood :)
I confirm that it could be a nice feature. We will investigate and let you know! 👍

**Reply by firegloves** [reply]
I'm not really sure that it is possible to implement this easily. I wasn't thinking that the query we are receiving could lack the limit statement. That means that MemPOI can't know how many rows will be exported... we may try to add a second query to express the count, but I'm not sure that it is a good way.

Do you have any idea?

**Reply by LuccaPrado** [reply]
I'm really sorry for the timing of my response. 

I don't have an idea about. I just thought that we can take the "where" from de query to do an count then keep the memPOI aligned with this number,  but, I guess that this wouldn't  work in all scenarios, right?
